guillotina/files/manager.py

```diff
@@ -231,26 +232,31 @@ def __init__(self, context, request, field):
                     )
                 await download_resp.write(chunk)
                 await download_resp.drain()
+        except Response:  # pragma: no cover
+            http_exception = True
+            raise
         except asyncio.CancelledError:  # pragma: no cover
             logger.info(f"Download cancelled: {self.request}")
             # when supporting range headers, the browser will
             # cancel downloads. This is fine.
             if download_resp is None:
+                http_exception = True
                 raise HTTPClientClosedRequest()
         finally:
-            if download_resp is None:
-                # deferred
-                download_resp = await self.prepare_download(
-                    disposition,
-                    filename,
-                    content_type,
-                    size,
-                    extra_headers,
-                    range_supported=range_supported,
-                    **kwargs,
-                )
-            await download_resp.write_eof()
-            return download_resp
+            if not http_exception:
```

Comment:
Also, this could be implemented using try...except...else so we only execute the following code if no exception ocurred

Reply:
yes, let me refactor.